src/main/java/racingcar/model/Cars.java

```diff
@@ -0,0 +1,40 @@
+package racingcar.model;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class Cars {
+    private List<Car> cars;
+
+    public Cars(List<Car> cars) {
+        validate(cars);
+        this.cars = cars;
+    }
+
+    public List<Car> getCars() {
+        return Collections.unmodifiableList(cars);
+    }
+
+    public void validate(List<Car> cars) {
+        validateCarCount(cars);
+        validateDuplication(cars);
+    }
+
+    private void validateDuplication(List<Car> cars) {
+        Set<String> duplicatedNames = new HashSet<>();
+        for (Car car : cars) {
+            duplicatedNames.add(car.getName());
+        }
+        if (duplicatedNames.size() != cars.size()) {
+            throw new IllegalArgumentException("이름은 중복이 불가합니다.");
+        }
```

Comment:
HashSet의 add는 존재값 추가할 때 false를 반환해요. 이를 통해 간소화할 수 있지 않을까요!
```suggestion
        for (Car car : cars) {
            if (duplicatedNames.add(car.getName())) {
                throw new IllegalArgumentException();
            }
        }
     
```

Reply:
add가 이미 존재하는 값에 대해 false를 반환하는지 처음 알았네요 !
감사합니다 :)